src/include/ipc/dai-intel.h

```diff
@@ -196,4 +197,27 @@ struct sof_ipc_dai_dmic_params {
 	struct sof_ipc_dai_dmic_pdm_ctrl pdm[0];
 } __attribute__((packed));
 
+struct sof_ipc_dai_multidai_ext {
+	uint32_t type;
+	uint32_t dai_index;
+
+	uint32_t reserved[6];
+	union {
+		struct sof_ipc_dai_ssp_params ssp;
+		struct sof_ipc_dai_dmic_params dmic;
+		struct sof_ipc_dai_hda_params hda;
+		struct sof_ipc_dai_alh_params alh;
+	};
+} __attribute__((packed));
+
+struct sof_ipc_dai_multidai_params {
+	struct sof_ipc_hdr hdr;
+	uint32_t num_dais;
+
+	uint32_t reserved[6];
+
+	struct sof_ipc_stream_map stream_map;
```

Comment:
We probably need a stream_map per dai config (since TDM modes can remap channels).

Reply:
Can you please elaborate what you mean by TDM mode channel remapping? This might be something entirely different,  so it might have to land in the SSP-specific IPC.

Reply:
In TDM mode, SSP can support upto 8 channels and these can be remapped to SSP slots differently from L/R mode. i.e. consider DAI config 0 is L/R mode stereo (slot 0,1), whilst DAI config 1 is 4 channel mode with L/R mapped to (slot 2,3).

Reply:
Yeah, that's SSP-specific, unrelated to the purpose of the map contained in multidai params.
It's for routing the stream channels to individual physical child DAIs.